core/pkg/launch/job_builder.go

```diff
@@ -333,10 +333,27 @@ func (j *JobBuilder) getEntrypoint(programPath string, metadata RunMetadata) ([]
 	if fullPython == nil {
 		return nil, fmt.Errorf("missing python attribute in metadata class")
 	}
+
+	// Extract just the Python version from metadata.Python.
+	// Format can be either "<python impl> <python version>" (new) or
+	// just "<python version>" (old).
+	pythonStr := *fullPython
+
+	// platform.python_implementation() returns ‘CPython’, ‘IronPython’,
+	// ‘Jython’, or ‘PyPy’, so we can split by whitespace.
+	parts := strings.Fields(pythonStr)
+	var pythonVersionSemver string
+
+	if len(parts) > 1 {
+		pythonVersionSemver = parts[len(parts)-1]
+	} else {
+		// Already just the version
+		pythonVersionSemver = pythonStr
+	}
+
 	// drop everything after the last second .
-	pythonVersion := strings.Join(strings.Split(*fullPython, ".")[:2], ".")
+	pythonVersion := strings.Join(strings.Split(pythonVersionSemver, ".")[:2], ".")
```

Comment:
This will panic if the `pythonVersionSemver` string doesn't have at least one period. Also this is a bit opaque: it's just a way of getting the `Major.Minor` form of the version, right? How about using a regex with capture groups to tackle validation + extraction at once? I think you want an expression like `(.+ )?(\d+\.\d+)(\.\d+)?`

If this can't figure out the version, should it return an error, or do something else?